### PR TITLE
Add ReplicaSelection to tokenBuilder

### PR DESCRIPTION
**Pull request by rbhanot**
ReplicaSelection as CLOSEST_REPLICA avoids querying leader tablet always and distribute the query load.
Fault tolerant queries allow scans to be retried to help avoid query failures when a tablet server containing the leaders goes down.

**Contributing to Twilio**

> All third-party contributors acknowledge that any contributions they provide will be made under the same open-source license that the open-source project is provided under.

- [ YES ] I acknowledge that all my contributions will be made under the project's license.

### CODE REVIEW

File: adapter/src/main/java/com/twilio/kudu/sql/KuduEnumerable.java [hunk 1]
```diff
@@ -487,6 +488,7 @@ public Schema getTableSchema() {
     List<AsyncKuduScanner> scanners = predicates.stream().map(subScan -> {
       KuduScanToken.KuduScanTokenBuilder tokenBuilder = client.syncClient()
           .newScanTokenBuilder(calciteKuduTable.getKuduTable());
+      tokenBuilder.replicaSelection(ReplicaSelection.CLOSEST_REPLICA); // Added to avoid querying leader always
```
Comment (minakshikorad) [inline adapter/src/main/java/com/twilio/kudu/sql/KuduEnumerable.java:491]: how do we know that leader is being queried all the time?

Comment (rbhanot) [reply adapter/src/main/java/com/twilio/kudu/sql/KuduEnumerable.java:491]: https://kudu.apache.org/apidocs/org/apache/kudu/client/AbstractKuduScannerBuilder.html 
```
Sets the replica selection mechanism for this scanner. The default is to read from the currently known leader.
```

The other option is to ensure we set `FaultTolerant` as true(which decreases throughput) and set READ_MODE as READ_AT_SNAPSHOT 
```
Fault tolerant scans use READ_AT_SNAPSHOT read mode. If no snapshot timestamp is provided, the server will pick one.
```

Comment (twdsilva) [reply adapter/src/main/java/com/twilio/kudu/sql/KuduEnumerable.java:491]: Do we need to set replica selection to `ReplicaSelection.CLOSEST_REPLICA` if we set fault tolerant to true?
Might be good to keep the comment about fault tolerant has to be set to true if we want rows returned in the pk order.

Comment (rbhanot) [reply adapter/src/main/java/com/twilio/kudu/sql/KuduEnumerable.java:491]: `ReplicaSelection.CLOSEST_REPLICA` will help us distribute our query traffic and is an independent change to the issue at hand.

Adding back the comment for rows returned in pk order.

**Review by twdsilva — APPROVED** [review]

**Pull request merged**